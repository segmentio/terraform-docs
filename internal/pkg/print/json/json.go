package json

import (
	"bytes"
	"encoding/json"
	"strings"

	"github.com/segmentio/terraform-docs/internal/pkg/print"
	"github.com/segmentio/terraform-docs/internal/pkg/tfconf"
)

// Print prints a document as json.
func Print(module *tfconf.Module, settings *print.Settings) (string, error) {
	module.Sort(settings)

	copy := &tfconf.Module{
		Header:    "",
		Providers: make([]*tfconf.Provider, 0),
		Inputs:    make([]*tfconf.Input, 0),
		Outputs:   make([]*tfconf.Output, 0),
	}

	if settings.ShowHeader {
		copy.Header = module.Header
	}
	if settings.ShowProviders {
		copy.Providers = module.Providers
	}
	if settings.ShowInputs {
		copy.Inputs = module.Inputs
	}
	if settings.ShowOutputs {
		copy.Outputs = module.Outputs
	}

	buffer := new(bytes.Buffer)

	encoder := json.NewEncoder(buffer)
	encoder.SetIndent("", "  ")
	encoder.SetEscapeHTML(settings.EscapeCharacters)

	err := encoder.Encode(copy)
	if err != nil {
		return "", err
	}

	return strings.TrimSuffix(buffer.String(), "\n"), nil
}
