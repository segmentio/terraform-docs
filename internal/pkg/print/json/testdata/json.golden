{
  "Comment": "Usage:\n\nmodule \"foo\" {\n  source = \"github.com/foo/bar\"\n\n  id   = \"1234567890\"\n  name = \"baz\"\n\n  zones = [\"us-east-1\", \"us-west-1\"]\n\n  tags = {\n    Name         = \"baz\"\n    Created-By   = \"first.last@email.com\"\n    Date-Created = \"20180101\"\n  }\n}\n\n",
  "Inputs": [
    {
      "Name": "string-2",
      "Description": "It's string number two.",
      "Default": null,
      "Type": "string"
    },
    {
      "Name": "string-1",
      "Description": "It's string number one.",
      "Default": {
        "Type": "string",
        "Literal": "bar"
      },
      "Type": "string"
    },
    {
      "Name": "map-2",
      "Description": "It's map number two.",
      "Default": null,
      "Type": "map"
    },
    {
      "Name": "map-1",
      "Description": "It's map number one.",
      "Default": {
        "Type": "map",
        "Literal": ""
      },
      "Type": "map"
    },
    {
      "Name": "list-2",
      "Description": "It's list number two.",
      "Default": null,
      "Type": "list"
    },
    {
      "Name": "list-1",
      "Description": "It's list number one.",
      "Default": {
        "Type": "list",
        "Literal": ""
      },
      "Type": "list"
    }
  ],
  "Outputs": [
    {
      "Name": "output-2",
      "Description": "It's output number two."
    },
    {
      "Name": "output-1",
      "Description": "It's output number one."
    }
  ]
}